src/AuthModule/Components/Register/Register.tsx

```diff
@@ -1,12 +1,398 @@
-import React from 'react'
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { ToastContainer, toast } from "react-toastify";
+import axios from "axios";
+import { RotatingLines } from "react-loader-spinner";
+import { SubmitHandler, FieldValues } from "react-hook-form";
+
+// Import components from MUI
+import {
+  Button,
+  FormControl,
+  TextField,
+  Typography,
+  FilledInput,
+  InputAdornment,
+  IconButton,
+  Alert,
+} from "@mui/material";
+import Visibility from "@mui/icons-material/Visibility";
+import VisibilityOff from "@mui/icons-material/VisibilityOff";
+
+// Import CSS module
+import styleRegister from "./Register.module.css";
+
+// Import image
+import imgRegister from "../../../Img/Group 33@2x.png";
+
+export interface FormDataRegister {
+  userName: string;
+  email: string;
+  country: string;
+  phoneNumber: string;
+  password: string;
+  confirmPassword: string;
+  role: string;
+  profileImage: FileList;
+}
 
 export default function Register() {
-  return (
+  const navigate = useNavigate();
+  const [loadingBtn, setLoadingBtn] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [dataApi, setdataApi] = useState([]);
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+    getValues,
+  } = useForm();
+
+  const validatePasswordMatch = (value: unknown) => {
+    const password = watch("password");
+    return value === password || "Confirm Password doesn't match Password"; 
+  };
+  const onSubmit: SubmitHandler<FormDataRegister> = async (data) => {
+    setLoadingBtn(true);
+
+    try {
+      const formData = new FormData();
+
+      formData.append("userName", data.userName);
+      formData.append("email", data.email);
+      formData.append("country", data.country);
+      formData.append("phoneNumber", data.phoneNumber);
+      formData.append("password", data.password);
+      formData.append("confirmPassword", data.confirmPassword);
+      formData.append("role", "user");
+      if (data.profileImage && data.profileImage[0]) {
+        formData.append("profileImage", data.profileImage[0]);
+      }
+
+      const response = await axios.post(
+        "https://upskilling-egypt.com:3000/api/v0/admin/users",
+        formData,
+      );
+
+      
+      console.log(response);
+      setdataApi(response.data);
+      toast.success("User created successfully");
+      navigate("/verifyAccount");
+    } catch (error:any) {
+      // console.log(error);
+
+      // console.log(error.response.data.message);
     
+      if (error.response.data.message == "email or userName  is already exists") {
```

Comment:
We need to re-check the error handling as it doesn't work properly.
We need to show the message coming from the backend and if not available provide a generic message such as  "something went wrong."